scripting.executeScript: result serialization incompatibility (#26857)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/scripting/executescript/index.md

Here the results array contains the string "`my result`" as an element.

> The result values must be [structured cloneable](/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm) in Firefox while [JSON serialization algorithm](/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description) is used in Chrome (see [Data cloning algorithm](/en-US/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities#data_cloning_algorithm) for more details on incompatibilities).

Reviewer: ```suggestion
The result values must be a [structured cloneable](/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm) in Firefox or [JSON serialization algorithm](/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description) object in Chrome. See [Data cloning algorithm](/en-US/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities#data_cloning_algorithm) in the Chrome incompatibilities article for more information).
```

Author: Thank you for review. I would avoid the term "object" since result may be a primitive value as a number or `true`/`false`.  Please, consider the following variant:

The script result must be a [structured cloneable](/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm) value in Firefox or a [JSON-serializable](/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description) value in Chrome. The [Chrome incompatibilities](/en-US/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities) article discusses it in more details in the [Data cloning algorithm](/en-US/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities#data_cloning_algorithm) section.

However I do not mind to use second sentence as in your suggestion (with removed closed parenthesis at the end).

Reviewer 2: sounds good to me

Author: I have added a commit with my variant

> The script result must be a [structured cloneable](https://github.com/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm) value in Firefox or a [JSON-serializable](https://github.com/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description) value in Chrome. The [Chrome incompatibilities](https://github.com/en-US/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities) article discusses it in more details in the [Data cloning algorithm](https://github.com/en-US/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities#data_cloning_algorithm) section.

Author: > @maxnikulin you seem to have introduced errors in the link URLs

Thanks. I am trying to get some impression of GitHub web UI. I did not expect that pasting formatted text may convert relative link hrefs to absolute ones.

> The script result must be a [structured cloneable](/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm) value in Firefox or a [JSON-serializable](/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description) value in Chrome. The [Chrome incompatibilities](/en-US/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities) article discusses this difference in more detail in the [Data cloning algorithm](https://github.com/en-US/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities#data_cloning_algorithm) section.

Reviewer: ```suggestion
The script result must be a [structured cloneable](/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm) value in Firefox or a [JSON-serializable](/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description) value in Chrome. The [Chrome incompatibilities](/en-US/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities) article discusses this difference in more detail in the [Data cloning algorithm](/en-US/docs/Mozilla/Add-ons/WebExtensions/Chrome_incompatibilities#data_cloning_algorithm) section.
```

Reviewer: Sorry, missed on of the link issues.

Author: > Sorry, missed one of the link issues.

My bad, I did not noticed it as well
